http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeoKubernetesServiceInterp.scala

```diff
@@ -0,0 +1,412 @@
+      )
+    )
+
+    for {
+      nodepoolName <- KubernetesNameUtils.getUniqueName(NodepoolName.apply)
+    } yield Nodepool(
+      NodepoolLeoId(-1),
+      clusterId = clusterId,
+      nodepoolName,
+      status = NodepoolStatus.Precreating,
+      auditInfo,
+      machineType = machineConfig.machineType,
+      numNodes = machineConfig.numNodes,
+      autoscalingEnabled = machineConfig.autoscalingEnabled,
+      autoscalingConfig = Some(leoKubernetesConfig.nodepoolConfig.galaxyNodepoolConfig.autoscalingConfig),
+      List.empty,
+      List.empty
+    )
+  }
+
+  private[service] def getSaveableApp(googleProject: GoogleProject,
+                                      appName: AppName,
+                                      userInfo: UserInfo,
+                                      samResourceId: AppSamResourceId,
+                                      req: CreateAppRequest,
+                                      diskOpt: Option[PersistentDisk],
+                                      nodepoolId: NodepoolLeoId,
+                                      ctx: AppContext): Either[Throwable, SaveApp] = {
+    val now = ctx.now
+    val auditInfo = AuditInfo(userInfo.userEmail, now, None, now)
+
+    val allLabels = DefaultKubernetesLabels(googleProject,
+                                            appName,
+                                            userInfo.userEmail,
+                                            leoKubernetesConfig.serviceAccountConfig.leoServiceAccountEmail)
+      .toMap() ++ req.labels
+    for {
+      // check the labels do not contain forbidden keys
+      labels <- if (allLabels.contains(includeDeletedKey))
+        Left(IllegalLabelKeyException(includeDeletedKey))
+      else
+        Right(allLabels)
+      //galaxy apps need a disk
+      disk <- if (req.appType == AppType.Galaxy && diskOpt.isEmpty)
+        Left(AppRequiresDiskException(googleProject, appName, req.appType, ctx.traceId))
+      else Right(diskOpt)
+      namespaceName <- KubernetesName.withValidation(
+        s"${appName.value}-${leoKubernetesConfig.galaxyAppConfig.namespaceNameSuffix.value}",
+        NamespaceName.apply
+      )
+    } yield SaveApp(
+      App(
+        AppId(-1),
+        nodepoolId,
+        req.appType,
+        appName,
+        AppStatus.Precreating,
+        samResourceId,
+        auditInfo,
+        labels,
+        AppResources(
+          Namespace(
+            NamespaceId(-1),
+            namespaceName
+          ),
+          disk,
+          req.appType match {
+            case Galaxy =>
+              leoKubernetesConfig.galaxyAppConfig.services.map(config => KubernetesService(ServiceId(-1), config))
+          }
+        ),
+        List.empty
+      )
+    )
+  }
+
+}
+
+object LeoKubernetesServiceInterp {
+  private[service] def processDiskConfigRequest[F[_]: Parallel](
+    req: DiskConfigRequest,
+    googleProject: GoogleProject,
+    isDataproc: Boolean,
```

Comment:
this paramater isn't needed for this interp, but its used to illustrate what is needed to generalize is to other interps